### PR TITLE
Fix map sizes in GH+GRMHD FD code

### PR DESCRIPTION
**Pull request by nilsdeppe**
### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/Systems/GrMhd/GhValenciaDivClean/Tags.hpp [hunk 1]
```diff
@@ -70,8 +70,8 @@ using primitive_grmhd_reconstruction_tags =
 /// \brief Tags sent for spacetime evolution. All tags sent are the
 /// reconstruciton+spacetime tags
 using spacetime_reconstruction_tags =
-    tmpl::list<gr::Tags::SpacetimeMetric<3>, GeneralizedHarmonic::Tags::Phi<3>,
-               GeneralizedHarmonic::Tags::Pi<3>>;
+    tmpl::list<gr::Tags::SpacetimeMetric<3>, GeneralizedHarmonic::Tags::Pi<3>,
```
Comment (kidder) [inline src/Evolution/Systems/GrMhd/GhValenciaDivClean/Tags.hpp:73]: why is the order switched here?

Comment (nilsdeppe) [reply src/Evolution/Systems/GrMhd/GhValenciaDivClean/Tags.hpp:73]: This is now the correct order. I have (or will have, lost track) a PR that factors this order into a tmpl::list so different places don't need to by synced.

Comment (nilsdeppe) [reply src/Evolution/Systems/GrMhd/GhValenciaDivClean/Tags.hpp:73]: Derp, that's this PR. The list with Pi first is the order in the GH system and should be the one used...

Comment (nilsdeppe) [reply src/Evolution/Systems/GrMhd/GhValenciaDivClean/Tags.hpp:73]: Okay... I've refactored this changed into the first commit :D

**Review by kidder — APPROVED** [review]

**Pull request merged**